dumpling/export/dump.go

```diff
@@ -379,6 +400,83 @@ func (d *Dumper) dumpDatabases(tctx *tcontext.Context, metaConn *sql.Conn, taskC
 	return nil
 }
 
+// adjustDatabaseCollation adjusts db collation and return new create sql and collation
+func adjustDatabaseCollation(db *sql.Conn, parser1 *parser.Parser, originSQL string, dbName string) (string, string, error) {
+	stmt, err := parser1.ParseOneStmt(originSQL, "", "")
+	if err != nil {
+		return "", "", err
+	}
+	createStmt, ok := stmt.(*ast.CreateDatabaseStmt)
+	if !ok {
+		return originSQL, "", nil
+	}
+	for _, createOption := range createStmt.Options {
+		// already have 'Collation'
+		if createOption.Tp == ast.DatabaseOptionCollate {
+			return originSQL, createOption.Value, nil
+		}
+	}
+	// get db collation
+	collation, err := GetDBCollation(db, dbName)
```

Comment:
I suggest getting db collation in one time one sql and store them into a map instead of querying it every time.